compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/BIRGen.java

```diff
@@ -2833,7 +2833,7 @@ private void generateFPVarRef(BLangExpression fpVarRef, BInvokableSymbol funcSym
 
         setScopeAndEmit(
                 new BIRNonTerminator.FPLoad(fpVarRef.pos, funcSymbol.pkgID, funcName, lhsOp, params, new ArrayList<>(),
-                        funcSymbol.retType, funcSymbol.strandName, funcSymbol.schedulerPolicy));
+                        funcSymbol.type, funcSymbol.strandName, funcSymbol.schedulerPolicy));
```

Comment:
FPLoad seems to be expecting the return type though, right? https://github.com/ballerina-platform/ballerina-lang/blob/master/compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/model/BIRNonTerminator.java#L733

Since we have the params also, we can probably construct the function type by passing the flags also.

Wondering if the `returnType.tag != TypeTags.INVOKABLE` check in JvmInstructionGen L1578 was unnecessary to begin with.

Reply:
If we are using the function type, shouldn't we rename the field and maybe get rid of params too?

Reply:
Actually, it should be a function type. Even we pass the type earlier we haven't used it during codegen. Only place used is https://github.com/ballerina-platform/ballerina-lang/blob/master/compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/methodgen/LambdaGen.java#L427:L429

Based on the logic, it should still work. 

I will rename returnType -> type.

Reply:
IMO We can still keep the returnType.tag != TypeTags.INVOKABLE check to catch any missing cases in the future.